src/main.ts

```diff
@@ -31,6 +31,11 @@ export async function run() {
       let auth = !token ? undefined : `token ${token}`;
 
       const checkLatest = core.getBooleanInput('check-latest');
+
+      if (versionSpec === 'stable' || versionSpec === 'oldstable') {
```

Comment:
I think it is better to create enums or use a similar approach as for [setup-dotnet](https://github.com/actions/setup-dotnet/blob/main/src/setup-dotnet.ts#L8-L16).